ballerina/tests/xpath_test.bal

```diff
@@ -0,0 +1,284 @@
+// Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/test;
+
+@test:Config
+function emptyQueryTest() {
+    xml value = xml `<root></root>`;
+    string query = "//b";
+    () result = checkpanic transform(query, value);
+    test:assertEquals(result, ());
+}
+
+@test:Config
+function xmlValueTest() {
+    xml value = xml `<data><person><firstName>John</firstName><lastName>Doe</lastName></person></data>`;
+    string query = "/data/person";
+    xml result = checkpanic transform(query, value);
+    xmlEqual(result, xml `<person><firstName>John</firstName><lastName>Doe</lastName></person>`);
+}
+
+@test:Config
+function stringValueTest() {
+    xml value = xml `<root><name>John</name></root>`;
+    string query = "/root/name";
+    string result = checkpanic transform(query, value);
+    test:assertEquals(result, "John");
+}
+
+@test:Config
+function stringSingleTest() {
+    xml value = xml `<root><name>John</name></root>`;
+    string query = "/root/name";
+    "John" result = checkpanic transform(query, value);
+    test:assertEquals(result, "John");
+}
+
+@test:Config
+function stringUnionSingleTest() {
+    xml value = xml `<root><name>John</name></root>`;
+    string query = "/root/name";
+    "John"|"Jane" result = checkpanic transform(query, value);
+    test:assertEquals(result, "John");
+}
+
+@test:Config
+function stringSingleErrorTest() {
+    xml value = xml `<root><name>John</name></root>`;
+    string query = "/root/name";
```

Comment:
Can you check if this is the correct xpath to get the text of an xml element? Usually to get the text we do `/root/name/text()` and if we do `/root/name`, it would return the whole element (`<name>John</name>`)

Reply:
Yes from Saxon I am getting `<name>John</name>` then I extract the inner value ("John") and map it to string. My plan is for both `/root/name/text()` and `/root/name/` to be "John" (as long as the first one to always return a single value). Would you prefer only second one to do this?